content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -818,6 +818,74 @@ controller is tracking a Job using Pod finalizers by checking if the Job has the
 annotation `batch.kubernetes.io/job-tracking`. You should **not** manually add
 or remove this annotation from Jobs.
 
+### Pod Communication Within a Job Using Pod Hostnames
```

Comment:
Maybe we can have this as a new task page, similar to https://kubernetes.io/docs/tasks/job/indexed-parallel-processing-static/, and linked in [Job Patterns section](https://kubernetes.io/docs/concepts/workloads/controllers/job/#job-patterns)

Reply:
Makes sense, I migrated this to a new task page (following the format of the other job-related tasks), and linked to it from the job patterns section. 

I also have addressed all the other comments so this PR is ready for another look.